proxy/src/main/java/com/wavefront/agent/AbstractAgent.java

```diff
@@ -126,10 +126,13 @@ void initSslContext() throws SSLException {
       sslContext = SslContextBuilder.forServer(new File(proxyConfig.getPrivateCertPath()),
               new File(proxyConfig.getPrivateKeyPath())).build();
     }
-    if (proxyConfig.isEnableTLS()) {
+    if (!isEmpty(proxyConfig.getTlsPorts()) && sslContext == null) {
```

Comment:
do you need a conditional based on secureAllPorts here?

Reply:
This is **init** method. **secureAllPorts** will be derived from **tlsPorts** in the next line. This is just validation.

Reply:
kk